internal/alert/provider/email/send.go

```diff
@@ -0,0 +1,151 @@
+package email
+
+import (
+	"bytes"
+	"crypto/tls"
+	"fmt"
+	"math/rand"
+	"mime"
+	"mime/multipart"
+	"mime/quotedprintable"
+	"net"
+	"net/mail"
+	"net/smtp"
+	"net/textproto"
+	"strings"
+	"time"
+
+	"github.com/balerter/balerter/internal/alert/message"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+)
+
+func (e *Email) Send(message *message.Message) error {
+	var (
+		c    *smtp.Client
+		conn net.Conn
+		err  error
+	)
+	if e.conf.ServerPort == "465" {
+		tlsConfig := &tls.Config{InsecureSkipVerify: true}
+
+		if tlsConfig.ServerName == "" {
+			tlsConfig.ServerName = e.conf.ServerName
+		}
+
+		conn, err = tls.Dial("tcp", e.conf.ServerName, tlsConfig)
+		if err != nil {
+			return errors.Wrap(err, "establish TLS connection to server")
+		}
+	} else {
+		var (
+			d   = net.Dialer{}
+			err error
+		)
+		conn, err = d.Dial("tcp", net.JoinHostPort(e.conf.ServerName, e.conf.ServerPort))
+		if err != nil {
+			return errors.Wrap(err, "establish connection to server")
```

Comment:
I think, let's use fmt.Errorf("... %w", err) for consistency

Reply:
Done.